src/Nethermind/Nethermind.Merge.Plugin/Handlers/V1/NewPayloadV1Handler.cs

```diff
@@ -273,64 +264,65 @@ private bool ShouldProcessBlock(Block block, BlockHeader parent, out ProcessingO
             return parentProcessed || processTerminalBlock;
         }
 
-        private async Task<(ValidationResult, string? Message)> ValidateBlockAndProcess(Block block, BlockHeader parent, ProcessingOptions processingOptions)
+        private async Task<(ValidationResult, string?)> ValidateBlockAndProcess(Block block, BlockHeader parent, ProcessingOptions processingOptions)
         {
-            ValidationResult ToValid(bool valid) => valid ? ValidationResult.Valid : ValidationResult.Invalid;
-            string? validationMessage = null;
+            ValidationResult CacheResult(ValidationResult result)
+            {
+                // notice that it is not correct to add information to the cache
+                // if we return SYNCING for example, and don't know yet whether
+                // the block is valid or invalid because we haven't processed it yet
+                if (result != ValidationResult.Syncing)
+                    _latestBlocks?.Set(block.GetOrCalculateHash(), result == ValidationResult.Valid);
+                return result;
+            }
+
+            (ValidationResult? result, string? validationMessage) = (null, null);
 
             // If duplicate, reuse results
-            bool isRecentBlock = _latestBlocks.TryGet(block.Hash!, out bool isValid);
-            if (isRecentBlock)
+            if (_latestBlocks is not null && _latestBlocks.TryGet(block.Hash!, out bool isValid))
             {
-                if (!isValid && _logger.IsWarn)
+                if (!isValid)
                 {
-                    validationMessage = $"Invalid block found in latestBlock cache.";
+                    validationMessage = "Invalid block found in latestBlock cache.";
                     if (_logger.IsWarn) _logger.Warn(validationMessage);
                 }
 
-                return (ValidationResult.AlreadyKnown | ToValid(isValid), validationMessage);
+                return (isValid ? ValidationResult.Valid : ValidationResult.Invalid, validationMessage);
```

Comment:
Can we cache syncing and later return it as invalid?

Reply:
We are never caching Syncing